eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -556,13 +556,21 @@ object Router {
    */
   case class PaymentContext(id: UUID, parentId: UUID, paymentHash: ByteVector32)
 
-  case class Route(amount: MilliSatoshi, hops: Seq[ChannelHop]) {
-    require(hops.nonEmpty, "route cannot be empty")
+  case class Route(amount: MilliSatoshi, hops: Seq[ChannelHop], finalHop_opt: Option[FinalHop]) {
+    require(hops.nonEmpty || finalHop_opt.nonEmpty, "route cannot be empty")
+
+    /** Full route including the final hop, if any. */
+    val fullRoute: Seq[Hop] = hops ++ finalHop_opt.toSeq
 
-    val length = hops.length
+    /**
+     * Fee paid for the trampoline hop, if any.
+     * Note that when using MPP to reach the trampoline node, the trampoline fee must be counted only once.
+     */
+    val trampolineFee: MilliSatoshi = finalHop_opt.collect { case hop: NodeHop => hop.fee(amount) }.getOrElse(0 msat)
```

Comment:
Right, but I think it's important to have it here to show that it's not covered by `channelFee`...these fee functions are a bit dangerous to use because `includeLocalChannelCost` doesn't work in all cases for blinded routes, that's why I wanted to decompose them explicitly to make sure callers beware.